try catch trycatch test no exception testnoexception foo foo exception